cohttp-eio/src/cohttp_eio.mli

```diff
@@ -96,3 +100,80 @@ module Server : sig
 
   val not_found_handler : handler
 end
+
+module Client : sig
+  type response = Http.Response.t * Eio.Buf_read.t
+
+  type host = string * int option
+  (** Represents a server host - as ip address or domain name - and an optional
+      port value, e.g. www.example.org:8080, www.reddit.com *)
+
+  type resource_path = string
+  (** Represents HTTP request resource path, e.g. "/shop/purchase",
+      "/shop/items", "/shop/categories/" etc. *)
+
+  type ('a, 'b) conn = 'a -> (resource_path * host * #Eio.Flow.two_way as 'b)
+  (** [('a, 'b conn)] is [(resource_path, host, flow)]. [flow] is the Eio flow
+      value which is connected to the [host]. *)
+
+  type ('a, 'b) body_disallowed_call =
+    ?version:Http.Version.t ->
+    ?headers:Http.Header.t ->
+    ('a, 'b) conn ->
+    'a ->
+    response
+  (** [body_disallowed_call] denotes HTTP client calls where a request is not
+      allowed to have a request body. *)
+
+  type ('a, 'b) body_allowed_call =
+    ?version:Http.Version.t ->
+    ?headers:Http.Header.t ->
+    ?body:Body.t ->
+    ('a, 'b) conn ->
+    'a ->
+    response
+  (** [body_allowed_call] denotes HTTP client calls where a request can
+      optionally have a request body. *)
+
+  (** {1 Generic HTTP call} *)
+
+  val call :
+    ?meth:Http.Method.t ->
+    ?version:Http.Version.t ->
+    ?headers:Http.Header.t ->
+    ?body:Body.t ->
+    ('a, 'b) conn ->
+    'a ->
```

Comment:
```suggestion
    'a conn ->
```
There is no point taking a function and its input here, as you just call it once immediately in all cases. It should just take the connection directly.

Reply:
Done in 7599e46254af78cc176d7a982deb3815d95a191a